.travis.yml

```diff
@@ -1,93 +1,44 @@
 ---
-language: generic
-
-addons:
-  apt:
-    packages:
-    - gettext
-    - libcurl4-openssl-dev
-    - libicu-dev
-    - libssl-dev
-    - libunwind8
-    - zlib1g
+language: csharp
+solution: mono/MaxMind.Db.Mono.sln
 
 env:
   global:
-    - MAXMIND_BENCHMARK_DB=MaxMind.Db.Benchmark/GeoLite2-City.mmdb
-    - MAXMIND_TEST_BASE_DIR=MaxMind.Db.Test
+    - MAXMIND_BENCHMARK_DB=$TRAVIS_BUILD_DIR/MaxMind.Db.Benchmark/GeoLite2-City.mmdb
+    - MAXMIND_TEST_BASE_DIR=$TRAVIS_BUILD_DIR/MaxMind.Db.Test
 
 matrix:
   include:
-    - os: linux
-      dist: trusty # Ubuntu 14.04
-      sudo: required
-      env: CONFIGURATION=Debug
-    - os: linux
-      dist: trusty
-      sudo: required
-      env: CONFIGURATION=Release
-    - os: osx
-      osx_image: xcode7.2 # macOS 10.11
-      env: CONFIGURATION=Debug
-    - os: osx
-      osx_image: xcode7.2
-      env: CONFIGURATION=Release
-
-before_install:
-  # Install OpenSSL
-  # Also set download URLs obtained from https://www.microsoft.com/net/download#core
-  - if test "$TRAVIS_OS_NAME" == "osx"; then
-      brew install openssl;
-      brew link --force openssl;
-      export DOTNET_SDK_URL="https://go.microsoft.com/fwlink/?LinkID=809128";
-    else
-      export DOTNET_SDK_URL="https://go.microsoft.com/fwlink/?LinkID=809129";
-    fi
-
-  - export DOTNET_INSTALL_DIR="$PWD/.dotnetcli"
-
-  # Install .NET CLI
-  - mkdir $DOTNET_INSTALL_DIR
-  - curl -L $DOTNET_SDK_URL -o dotnet_package
-  - tar -xvzf dotnet_package -C $DOTNET_INSTALL_DIR
-
-  # Add dotnet to PATH
-  - export PATH="$DOTNET_INSTALL_DIR:$PATH"
-
-  # Update submodules
-  - git submodule update --init --recursive
-
-install:
-  # Display dotnet version info
-  - which dotnet;
-    if [ $? -eq 0 ]; then
-      echo "Using dotnet:";
-      dotnet --info;
-    else
-      echo "dotnet.exe not found"
-      exit 1;
-    fi
-
-  # Hack for: https://github.com/dotnet/cli/issues/3658
-  - if test "$TRAVIS_OS_NAME" == "linux"; then
-      nvm install stable && nvm use stable;
-    fi
-  - node -e "jsonPath='./MaxMind.Db.Benchmark/project.json';data=require(jsonPath);fs=require('fs');delete data.frameworks.net45;fs.writeFileSync(jsonPath, JSON.stringify(data, null, 2))"
-
-  # Restore Packages
-  - dotnet restore
-
-  # Build Projects
-  - dotnet build -c $CONFIGURATION -f netstandard1.4 ./MaxMind.Db
-  - dotnet build -c $CONFIGURATION -f netcoreapp1.0 ./MaxMind.Db.Benchmark
-  - dotnet build -c $CONFIGURATION -f netcoreapp1.0 ./MaxMind.Db.Test
-
-script:
-  # Run Benchmark
-  - dotnet run -f netcoreapp1.0 -c $CONFIGURATION -p MaxMind.Db.Benchmark
-
-  # Run Unit Tests
-  - dotnet test -f netcoreapp1.0 -c $CONFIGURATION MaxMind.Db.Test
+  - os: linux
+    dist: trusty # Ubuntu 14.04
+    sudo: required
+    dotnet: 1.0.0-preview2-003121
+    mono: none
+    env: DOTNETCORE=1 CONFIGURATION=Release
+    sudo: required
```

Comment:
You also have `sudo` twice here but as `required` both times. Is it really required? Builds are generally slower with it enabled.